meteor/client/ui/Prompter/PrompterView.tsx

```diff
@@ -653,46 +657,134 @@ export const Prompter = translateWithTracker<PropsWithChildren<IPrompterProps>,
+			/** Maps anchorId -> offset */
+			const foundAnchors = new Map<
+				string,
+				{
+					/** offset of the anchor compared to current viewport (0 = top of the viewport) */
+					offset: number
+					/** Positive number. How "good" the anchor is. The anchor with the lowest number will preferred later. */
+					distanceToReadPosition: number
+				}
+			>()
+
+			/*
+				The idea here is to find the best anchor we can use to tie the prompter scroll position to the viewed content.
+			*/
 
 			const windowInnerHeight = window.innerHeight
-			const margin = windowInnerHeight / 30
 
-			Array.from(document.querySelectorAll('.prompter .prompter-line:not(.empty)')).forEach((anchor) => {
-				const { top, bottom } = anchor.getBoundingClientRect()
-				// find a prompter line that is in the read area of the viewport
-				if (top <= windowInnerHeight && bottom >= readPosition) foundPositions.push([top, anchor.id])
-			})
+			// Gather anchors from the text blocks in view
+			{
+				const textAnchors = Array.from(document.querySelectorAll('.prompter .prompter-line:not(.empty)'))
+					.map((anchor) => {
+						const { top, bottom } = anchor.getBoundingClientRect()
+						return { id: anchor.id, top, bottom }
+					})
+					.sort((a, b) => a.top - b.top)
 
-			// if we didn't find any text, let's use scroll-anchors instead (Segment and Part names)
-			if (foundPositions.length === 0) {
-				Array.from(document.querySelectorAll('.prompter .scroll-anchor')).forEach((anchor) => {
-					const { top } = anchor.getBoundingClientRect()
-					// find a prompter scroll anchor that is just above the prompter read position
-					if (top <= readPosition + margin) foundPositions.push([top, anchor.id])
-				})
+				for (const anchor of textAnchors) {
+					const meanPosition = (anchor.top + anchor.bottom) / 2
+					const distanceToReadPosition = Math.abs(meanPosition - readPosition)
+
+					// Is the text block in view?
+					if (anchor.top <= windowInnerHeight && anchor.bottom > 0) {
+						foundAnchors.set(anchor.id, {
+							offset: anchor.top,
+
+							// Multiply by a small number to make these have higher priority in the sorting later.
+							// (We want to prefer these over the scroll-anchors)
+							distanceToReadPosition: distanceToReadPosition * 0.0001,
+						})
+					}
+				}
 			}
 
-			foundPositions = _.sortBy(foundPositions, (topOffset) => Number(topOffset[0]))
+			// Also use scroll-anchors (Segment and Part names)
+			{
+				const scrollAnchors = Array.from(document.querySelectorAll('.prompter .scroll-anchor'))
+					.map((anchor) => {
+						const { top, bottom } = anchor.getBoundingClientRect()
+						return { id: anchor.id, top, bottom }
+					})
+					.sort((a, b) => a.top - b.top)
+				for (const anchor of scrollAnchors) {
+					// find a prompter scroll anchor that is just above the prompter read position:
+					const distanceToReadPosition = Math.abs(anchor.top - readPosition)
```

Comment:
I think this is too generic. The fact is we don't want to just find the closest match, we want to maintain whatever the reader is most likely focusing on, which may in fact not be the one with it's `top` being the closest. Consider a situation where one is reading the very end of a very long paragraph in Part A, and seeing Part B in view port. Part A's `anchor.top` will in fact be further away than Part B's `anchor.top`, But we want to keep Part A static, and Part B can just move around, if for example the editor moves a new Part C in front of Part B.